src/db_utils.py

```diff
@@ -0,0 +1,133 @@
+"""
+Copyright 2023 NOAA
+All rights reserved.
+
+Collection of methods to facilitate handling of score db requests
+
+"""
+
+from collections import namedtuple
+import copy
+from dataclasses import dataclass, field
+from datetime import datetime
+import json
+import pprint
+from db_action_response import DbActionResponse
+import score_table_models as stm
+from score_table_models import MetricType as mt
+import time_utils
+
+from pandas import DataFrame
+import sqlalchemy as db
+from sqlalchemy.dialects.postgresql import insert
+from sqlalchemy.inspection import inspect
+from sqlalchemy import and_, or_, not_
+from sqlalchemy import asc, desc
+from sqlalchemy.sql import func
+
+ASCENDING = 'asc'
+DESCENDING = 'desc'
+
+VALID_ORDER_BY = [ASCENDING, DESCENDING]
+
+HTTP_GET = 'GET'
+HTTP_PUT = 'PUT'
+
+VALID_METHODS = [HTTP_GET, HTTP_PUT]
+
+INSERT = 'INSERT'
+UPDATE = 'UPDATE'
+
+EXACT_DATETIME = 'exact'
+FROM_DATETIME = 'from'
+TO_DATETIME = 'to'
+
+HERA = 'hera'
+ORION = 'orion'
+PW_AZV1 = 'pw_azv1'
+PW_AZV2 = 'pw_azv2'
+PW_AWV1 = 'pw_awv1'
+PW_AWV2 = 'pw_awv2'
+
+VALID_PLATFORMS = [HERA, ORION, PW_AZV1, PW_AZV2, PW_AWV1, PW_AWV2]
+
+def validate_column_name(cls, value):
+    if not isinstance(value, str):
+        raise TypeError(f'Column name must be a str, was {type(value)}')
+
+    try:
+        column_obj = getattr(cls, value)
+        print(f'column: {column_obj}, type(key): {type(column_obj)}')
+    except Exception as err:
+        msg = f'Column does not exist - err: {err}'
+        raise ValueError(msg)
```

Comment:
As discussed in person, this would be a much larger change then intended for this PR for the structure of the repo and as written, currently is effective for debugging. This is a change to consider in future as part of a larger rewrite of the repo if desired.